plugins/modules/pfsense_virtual_ip.py

```diff
@@ -0,0 +1,316 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: pfsense_virtual_ip
+version_added: "0.6.2"
+author: Jan Wenzel (@coffeelover)
+short_description: Manage pfSense virtual ip settings
+description:
+  - Manage pfSense virtual ip settings
+notes:
+options:
+  mode:
+    description: Type
+    required: True
+    type: str
+    choices: [ "proxyarp", "carp", "ipalias", "other" ]
+  descr:
+    description: Description
+    required: False
+    type: str
+  interface:
+    description: Interface
+    required: True
+    type: str
+  vhid:
+    description: VHID Group
+    required: False
+    type: int
+  advbase:
+    description: Advertising Frequency Base
+    required: False
+    type: int
+  advskew:
+    description: Advertising Frequency Skew
+    required: False
+    type: int
+  password:
+    description: Virtual IP Password
+    required: False
+    type: str
+  uniqid:
+    description: Unique ID of Virtual IP in configuration
+    required: False
+    type: str
+  type:
+    description: Address Type
+    required: False
+    type: str
+    choices: [ "single" ]
+    default: single
+  subnet_bits:
+    description: Network's subnet mask
+    required: False
+    type: int
+    default: 32
+  subnet:
+    description: Network subnet
+    required: False
+    type: str
+  state:
+    description: State in which to leave the Virtual IP
+    choices: [ "present", "absent" ]
+    default: present
+    type: str
+"""
+
+EXAMPLES = """
+- name: Setup Home vip
+  pfsense_virtual_ip:
+    mode: "carp"
+    descr: "HOME VIP"
+    interface: "opt2"
+    vhid: 24
+    advbase: 1,
+    advskew: 0,
+    password": "xaequae0sheiB7sh"
+    uniqid": "vip_home"
+    subnet_bits": 24
+    subnet": "10.1.1.1"
+    state": "present"
+"""
+
+RETURN = """
+"""
+
+import re
+from copy import deepcopy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.pfsensible.core.plugins.module_utils.module_base import PFSenseModuleBase
+
+
+VIRTUALIP_ARGUMENT_SPEC = dict(
+    mode=dict(required=True, choices=['proxyarp', 'carp', 'ipalias', 'other'], type='str'),
+    interface=dict(required=True, type='str'),
+    vhid=dict(required=False, type='int'),
+    advskew=dict(required=False, type='int'),
+    advbase=dict(required=False, type='int'),
+    password=dict(required=False, type='str', no_log=True),
+    uniqid=dict(required=False, type='str'),
+    descr=dict(required=False, type='str'),
+    type=dict(required=False, type='str', choices=['single'], default='single'),
+    subnet_bits=dict(required=False, type='int', default=32),
+    subnet=dict(required=False, type='str'),
+    state=dict(default='present', choices=['present', 'absent'], type='str'),
+)
+
+VIRTUALIP_REQUIRED_IF = [
+    ["mode", "carp", ["uniqid", "password", "vhid", "advbase"]],
```

Comment:
I would think we would need a `"state", "present"` entry.

Reply:
Hmm, and an absent ?

Reply:
It is needed for both states.